### PR TITLE
[R] Handling null values for optional params while building the model json object

### PR DESCRIPTION
**Pull request by Ramanth**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh`, `./bin/security/{LANG}-petstore.sh` and `./bin/openapi3/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
**Error when optional parameters are not given for the model object**

When generating the json of a model using the toJSONString() we encounter an error when we dont pass the optional paramameters (which are of non-primitive type)

Below changes handles the listContainer type optional parameters by replacing them with "null". so that it will be ignored by the endpoint.

Please make necessary changes if required.
@wing328 , please look into and take required action.

**Example to test:**
please consider a model which is hanving an example as below
```
{
	"key1" : "value1",
	"key2" : [  {"k1" : "v1"},
                    {"k2" : "v2"} ],
	"key3" : {  "k3" : "v3",
                     "k4" : "v4"
         	      }
}
```

In the above example consider key2 as an optional parameter and of type listContainer (i.e, array). 
Which is throwing an error when we dont provide the value for key2 while generating the model object.
Please review this changes and make required changes to handle such scenarios.

Thanks,
Ramanth.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/r/model.mustache [hunk 1]
```diff
@@ -181,24 +181,32 @@
...
+        jsonlite::toJSON(self$`{{baseName}}`$toJSON(), auto_unbox=TRUE)
         {{/isPrimitiveType}}
         {{/isListContainer}}
+	}
+	else{
+	  "null"
+	}{{#hasMore}},{{/hasMore}}
         {{/vars}}
       )
+	  outstring = gsub("[\r\n]| ", "", outstring)
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/r/model.mustache]: I'm afraid this won't work if the data/value contains line breaks

Comment (Ramanth) [reply modules/openapi-generator/src/main/resources/r/model.mustache]: @wing328 you are right, i can remove this line considering what you mentioned. Thanks

**Comment by wing328** [comment]
As discussed, instead of using `sprintf`, what about using string concatenation instead so that we can check for null before adding anything to the output?

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/pull/2777 addresses this problem for Dart2. We should adopt a similar approach for R

**Comment by Ramanth** [comment]
raised a new [PR](https://github.com/OpenAPITools/openapi-generator/pull/3082) with different approach.

**Pull request closed without merging**